File: tests/samples/sample.py
#!/usr/bin/env python3
"""sample.py - synthetic Python file for Explorer preview testing."""

from dataclasses import dataclass


@dataclass
class Item:
    id: int
    label: str


def describe(item: Item) -> str:
    return f"#{item.id}: {item.label}"


def main() -> None:
    items = [Item(1, "Apple"), Item(2, "Banana"), Item(3, "Grape")]
    for item in items:
        print(describe(item))


if __name__ == "__main__":
    main()

File: tests/samples/sample.ts
// sample.ts - synthetic TypeScript file for Explorer preview testing

interface Item {
  id: number;
  label: string;
}

const items: Item[] = [
  { id: 1, label: "Apple" },
  { id: 2, label: "Banana" },
  { id: 3, label: "Grape" },
];

function describe(item: Item): string {
  return `#${item.id}: ${item.label}`;
}

items.forEach((item) => console.log(describe(item)));
